ion-schema/src/result/invalid_schema_error.rs

```diff
@@ -0,0 +1,165 @@
+            InvalidSchemaErrorQuantity::One(error) => {
+                self.errors.insert(error);
+            }
+        }
+    }
+
+    /// Converts from [`IonSchemaResult<T>`] to [`Option<T>`].
+    ///
+    /// Converts `result` into an [`Option<T>`], consuming `result`, and pushing the error, if any,
+    /// to this `ErrorCollector`.
+    /// This is the multi-error equivalent to [`Result::ok`].
+    pub(crate) fn ok_or_push_err<T>(
+        &mut self,
+        result: impl Into<Result<T, InvalidSchemaError>>,
+    ) -> Option<T> {
+        match result.into() {
+            Ok(value) => Some(value),
+            Err(e) => {
+                self.push_err(e);
+                None
+            }
+        }
+    }
+}
+
+#[derive(Debug, Error, PartialEq)]
+#[error(transparent)]
+pub struct InvalidSchemaError(InvalidSchemaErrorQuantity);
+
+impl InvalidSchemaError {
+    pub(crate) fn builder(description: String) -> InvalidSchemaDetailsBuilder {
+        InvalidSchemaDetailsBuilder::new(description)
+    }
+}
+
+pub struct InvalidSchemaDetailsBuilder {
+    schema_id: String,
+    row_column: IslSourceLocation,
+    description: String,
+}
+
+impl InvalidSchemaDetailsBuilder {
+    fn new(description: String) -> Self {
+        InvalidSchemaDetailsBuilder {
+            description,
+            row_column: IslSourceLocation::default(),
+            schema_id: "".to_string(),
+        }
+    }
+    pub(crate) fn with_schema_id(mut self, id: impl Into<String>) -> Self {
+        self.schema_id = id.into();
+        self
+    }
+    pub(crate) fn with_isl_source_location<T: HasIslSourceLocation>(mut self, item: &T) -> Self {
+        self.row_column = item.isl_source_location();
+        self
+    }
+    pub(crate) fn build(self) -> InvalidSchemaError {
+        InvalidSchemaError(InvalidSchemaErrorQuantity::One(InvalidSchemaDetails {
+            schema_id: self.schema_id,
+            row_column: self.row_column,
+            description: self.description,
+        }))
+    }
+}
+impl From<InvalidSchemaDetailsBuilder> for InvalidSchemaError {
+    fn from(value: InvalidSchemaDetailsBuilder) -> Self {
+        value.build()
+    }
+}
+impl<T> From<InvalidSchemaDetailsBuilder> for Result<T, InvalidSchemaError> {
+    fn from(value: InvalidSchemaDetailsBuilder) -> Self {
+        Err(value.build())
+    }
+}
+impl From<InvalidSchemaDetailsBuilder> for IonSchemaError {
+    fn from(value: InvalidSchemaDetailsBuilder) -> Self {
+        IonSchemaError::InvalidSchemaError(value.build())
+    }
+}
+impl<T> From<InvalidSchemaDetailsBuilder> for IonSchemaResult<T> {
+    fn from(value: InvalidSchemaDetailsBuilder) -> Self {
+        Err(IonSchemaError::InvalidSchemaError(value.build()))
+    }
+}
+
+#[derive(Debug, Error, PartialEq)]
+enum InvalidSchemaErrorQuantity {
+    One(InvalidSchemaDetails),
+    Many(BTreeSet<InvalidSchemaDetails>),
+}
+impl Display for InvalidSchemaErrorQuantity {
```

Comment:
❗ When there are multiple errors in an `InvalidSchemaError`, they are displayed like this:
```
foo/bar.isl:24:13 > range start must not be greater than range end
foo/bar.isl:105:22 > top level type missing name
foo/baz.isl:1:1 > unknown schema version 3.14
```

Reply:
Love this new `Display` for error reporting!